src/bentoml/_internal/runner/utils.py

```diff
@@ -152,3 +154,28 @@ def payload_paramss_to_batch_params(
             f"argument lengths for parameters do not matchs: {tuple(indice_params.items())}"
         )
     return batched_params, indice_params.sample
+
+
+# This is a reference to starlette.concurrency
+class _StopIteration(Exception):
+    pass
+
+
+def _next(iterator: t.Iterator[T]) -> T:
+    # We can't raise `StopIteration` from within the threadpool iterator
+    # and catch it outside that context, so we coerce them into a different
+    # exception type.
+    try:
+        return next(iterator)
+    except StopIteration:
+        raise _StopIteration
+
+
+async def iterate_in_threadpool(
+    iterator: t.Iterator[T],
+) -> t.AsyncIterator[T]:
+    while True:
+        try:
+            yield await anyio.to_thread.run_sync(_next, iterator)
+        except _StopIteration:
```

Comment:
Do you know why it is using a custom error instead of `StopIteration`?

Reply:
From what I gathered, [PEP 479](https://peps.python.org/pep-0479/) will bubble `RuntimeError` up the context when `StopIteration` is raised within a generator. Hence, to make things clearer, we used a custom error here.